dev/TeachingTip/TeachingTip.cpp

```diff
@@ -844,38 +919,61 @@ void TeachingTip::OnIsOpenChanged()
                 }
             }
         }
-
-        m_acceleratorKeyActivatedRevoker = Dispatcher().AcceleratorKeyActivated(winrt::auto_revoke, { this, &TeachingTip::OnF6AcceleratorKeyClicked });
     }
-    else
+
+    m_acceleratorKeyActivatedRevoker = Dispatcher().AcceleratorKeyActivated(winrt::auto_revoke, { this, &TeachingTip::OnF6AcceleratorKeyClicked });
+}
+
+void TeachingTip::IsOpenChangedToClose()
+{
+    if (auto&& popup = m_popup.get())
     {
-        if (auto&& popup = m_popup.get())
+        if (popup.IsOpen())
         {
-            if (popup.IsOpen())
+            // We are about to begin the process of trying to close the teaching tip, so notify that we are no longer idle.
+            if (m_isIdle)
             {
-                // We are about to begin the process of trying to close the teaching tip, so notify that we are no longer idle.
-                if (m_isIdle)
-                {
-                    m_isIdle = false;
-                    TeachingTipTestHooks::NotifyIdleStatusChanged(*this);
-                }
-                RaiseClosingEvent(true);
+                m_isIdle = false;
+                TeachingTipTestHooks::NotifyIdleStatusChanged(*this);
             }
-            else
+            RaiseClosingEvent(true);
+        }
+        else
+        {
+            // We have become not Open but our popup was already not open. Lets make sure the idle status is correct.
+            if (!m_isIdle && !m_isExpandAnimationPlaying && !m_isContractAnimationPlaying)
             {
-                // We have become not Open but our popup was already not open. Lets make sure the idle status is correct.
-                if (!m_isIdle && !m_isExpandAnimationPlaying && !m_isContractAnimationPlaying)
-                {
-                    m_isIdle = true;
-                    TeachingTipTestHooks::NotifyIdleStatusChanged(*this);
-                }
+                m_isIdle = true;
+                TeachingTipTestHooks::NotifyIdleStatusChanged(*this);
             }
         }
+    }
 
-        m_currentEffectiveTipPlacementMode = winrt::TeachingTipPlacementMode::Auto;
-        TeachingTipTestHooks::NotifyEffectivePlacementChanged(*this);
+    m_currentEffectiveTipPlacementMode = winrt::TeachingTipPlacementMode::Auto;
+    TeachingTipTestHooks::NotifyEffectivePlacementChanged(*this);
+}
+
+void TeachingTip::CreateNewPopup()
+{
+    m_popupOpenedRevoker.revoke();
+    m_popupClosedRevoker.revoke();
+
+    auto const popup = winrt::Popup();
+    // Set XamlRoot on the popup to handle XamlIsland/AppWindow scenarios.
+    if (winrt::IUIElement10 uiElement10 = *this)
+    {
+        popup.XamlRoot(uiElement10.XamlRoot());
+    }
+
+    m_popupOpenedRevoker = popup.Opened(winrt::auto_revoke, { this, &TeachingTip::OnPopupOpened });
+    m_popupClosedRevoker = popup.Closed(winrt::auto_revoke, { this, &TeachingTip::OnPopupClosed });
+    if (auto&& popup3 = popup.try_as<winrt::Controls::Primitives::IPopup3>())
```

Comment:
>.try_as<winrt::Controls::Primitives::IPopup3>() [](start = 29, length = 47)

first, this shouldn't need to be namespaced if you alias it in CppWinrtincludes.h. Second, this pattern (as above with uiElement10) reads better as:

```
if (winrt::IPopup3 popup3 = popup)
{
...
}
```